.github/workflows/release.yml

```diff
@@ -0,0 +1,30 @@
+on:
+  workflow_call:
+    inputs:
+      cli-name:
+        required: true
+        type: string
+name: Release
+jobs:
+  release:
+    permissions:
+      id-token: write
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
```

Comment:
1.18 is the latest, how do we choose which version to use here? Is the default the latest release?
If we pin to a specific version here, how do we monitor for new versions?

Reply:
Go doesn't release that frequently so we should be on the safe side when we pin to a relatively new version. 

Anyway, figured out how to run the CI against all available go versions newer than the one we specify in the `go.mod` file, so that should significantly reduce the number of places where we pin go versions and thus always give a sense if we are okay against the latest one 👍

Reply:
The result is:

- Tests will run against every go version starting from the one we say we support in the `go.mod` file. 
- Releases will use the one we specify in the `go.mod` file